server/src/main/java/org/opensearch/snapshots/SnapshotsService.java

```diff
@@ -1769,10 +1770,49 @@ public ClusterState execute(ClusterState currentState) throws Exception {
                     snapshotNames,
                     repoName
                 );
-                deleteFromRepoTask = createDeleteStateUpdate(snapshotIds, repoName, repositoryData, Priority.NORMAL, listener);
+                final List<SnapshotId> snapshotIdNotBackingIndex = filterSnapshotsBackingAnyIndex(currentState, snapshotIds);
+                deleteFromRepoTask = createDeleteStateUpdate(
+                    snapshotIdNotBackingIndex,
+                    repoName,
+                    repositoryData,
+                    Priority.NORMAL,
+                    listener
+                );
                 return deleteFromRepoTask.execute(currentState);
             }
 
+            private List<SnapshotId> filterSnapshotsBackingAnyIndex(ClusterState currentState, List<SnapshotId> snapshotIds) {
+                final Set<SnapshotId> snapshotsToBeDeleted = new HashSet<>();
+                final Set<String> snapshotsToBeNotDeleted = new HashSet<>();
+                ImmutableOpenMap<String, IndexMetadata> indicesMap = currentState.getMetadata().getIndices();
+                for (SnapshotId snapshotId : snapshotIds) {
+                    Boolean indexBackedBySnapshotFound = false;
+                    for (Iterator<IndexMetadata> it = indicesMap.valuesIt(); it.hasNext();) {
+                        IndexMetadata indexMetadata = it.next();
+                        String storeType = indexMetadata.getSettings().get(IndexModule.INDEX_STORE_TYPE_SETTING.getKey());
+                        if (storeType != null
```

Comment:
You can instead do `IndexModule.Type.REMOTE_SNAPSHOT.getSettingsKey().equals(storeType)` and get rid of the `null` check.